candle-transformers/src/models/llama.rs

```diff
@@ -225,8 +225,10 @@ impl CausalSelfAttention {
 
         if cache.use_kv_cache {
             if let Some((cache_k, cache_v)) = &cache.kvs[block_idx] {
-                k = Tensor::cat(&[cache_k, &k], 2)?.contiguous()?;
-                v = Tensor::cat(&[cache_v, &v], 2)?.contiguous()?;
+                // k = Tensor::cat(&[cache_k, &k], 2)?.contiguous()?;
+                // v = Tensor::cat(&[cache_v, &v], 2)?.contiguous()?;
+                k = candle_nn::kvconcat(&cache_k, &k, 2)?;
```

Comment:
Perhaps this could be implemented for the quantized_llama.rs example?

Reply:
Yes,  it should be workable on the quantized examples as well.